.gitlab/.pre/gitlab_configuration.yml

```diff
@@ -47,3 +50,18 @@ compute_gitlab_ci_config:
     paths:
       - artifacts/
     expire_in: 1 day
+
+lint_gitlab_ci_jobs_needs_rules:
+  stage: .pre
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  needs: [compute_gitlab_ci_config]
+  rules:
+    - !reference [.compute_gitlab_ci_config_rules]
+    - !reference [.on_gitlab_changes]
+    - changes:
+        paths:
+          - .ci-needs-rules-allowlist.yml
+        compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏I'm not a big fan of having this PR add 2 files at the root of the repo. Couldn't we use `.gitlab/.ci-linters.yml` and `.gitlab/.ci-needs-rules-allowlist.yml` instead ?